mcu-lib/mcu.py

```
@@ -0,0 +1,277 @@
+        packet = ReturnPacket(packet_data)
+        return packet
+
+    def __parse_packet(self, packet: ReturnPacket):
+        data_bs = packet.data[0] + packet.data[1] + packet.data[2] + packet.data[3]
+        if not packet:
+            return
+        # let's all pretend this was a Python 3.10+ match/case statement
+        if packet.cmd == bs(0x00):
+            # test
+            version = int.from_bytes(packet.data[0], 'big')
+            contents = (packet.data[1] + packet.data[2] + packet.data[3]).decode('latin')
+            valid = contents == "pog"
+            self.test_queue.put(TestPacket(valid, version, contents, packet.timestamp))
+        elif packet.cmd == bs(0x0A):
+            # OK
+            og_cmd = int.from_bytes(packet.og_cmd, 'big')
+            og_param = int.from_bytes(packet.og_param, 'big')
+            success = int.from_bytes(packet.param, 'big') > 0
+            self.ok_queue.put(OKPacket(og_cmd, og_param, success, packet.timestamp))
+        elif packet.cmd == bs(0x3A):
+            # accel
+            axis = int(int.from_bytes(packet.param, 'big') / 0x30)
+            value = struct.unpack('f', data_bs)
+            self.accel_queue.put(AccelPacket(axis, value, packet.timestamp))
+            self.latest_accel[axis] = value[0]
+        elif packet.cmd == bs(0x3C):
+            # gyro
+            axis = int(int.from_bytes(packet.param, 'big') / 0x30)
+            value = struct.unpack('f', data_bs)
+            self.gyro_queue.put(GyroPacket(axis, value, packet.timestamp))
+            self.latest_gyro[axis] = value[0]
+        elif packet.cmd == bs(0x44):
+            # temp/volt
+            temp, volts = struct.unpack('HH', data_bs)
+            temp /= 100
+            volts /= 100
+            self.latest_temp = temp
+            self.latest_voltage = volts
+            self.volt_temp_queue.put(VoltageTemperaturePacket(volts, temp, packet.timestamp))
+        else:
+            print("invalid packet")
```

Maybe raise an exception?

---

We shouldn't raise an exception here, as we want the program to continue running when it receives an invalid packet, possibly due to a bit-flip on the command.